src/Bicep.Core.Samples/Files/Resources_CRLF/main.bicep

```diff
@@ -215,4 +215,17 @@ resource resourceWithEscaping 'My.Rp/mockResource@2020-01-01' = {
     // both key and value should be escaped in template output
     '[resourceGroup().location]': '[resourceGroup().location]'
   }
+}
+
+param shouldDeployVm bool = true

```

Comment:
This could be interesting to have a case with runtime functions because the variable has to be inlined during codegen.

Reply:
Btw, I tested it with this file and it worked great!

```bicep
targetScope = 'managementGroup'

var shouldOrNot = reference('Microsoft.Management/managementGroups/test', '2020-05-01').details.parent.id == 'something'
resource foo 'Microsoft.Management/managementGroups@2020-05-01' = if (shouldOrNot) {
    name: 'test'
    properties: {
        details: {
            parent: {
            }
        }
    }
    dependsOn: []
}
```

Reply:
Confirmed that runtime functions are invalid inside resource conditions.